keyboards/ramonimbao/herringbone/pro/keymaps/via/keymap.c

```diff
@@ -0,0 +1,87 @@
+        _______, _______, _______,                   _______,                   _______, _______,                   _______, _______, _______
+    ),
+    [2] = LAYOUT_all(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,                   _______,                   _______, _______,                   _______, _______, _______
+    ),
+    [3] = LAYOUT_all(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,                   _______,                   _______, _______,                   _______, _______, _______
+    ),
+};
+
+keyevent_t encoder_ccw = {
+    .key = (keypos_t){.row = 6, .col = 0},
+    .pressed = false
+};
+
+keyevent_t encoder_cw = {
+    .key = (keypos_t){.row = 6, .col = 1},
+    .pressed = false
+};
+
+void matrix_scan_user(void) {
+    if (IS_PRESSED(encoder_ccw)) {
+        encoder_ccw.pressed = false;
+        encoder_ccw.time = (timer_read() | 1);
+        action_exec(encoder_ccw);
+    }
+
+    if (IS_PRESSED(encoder_cw)) {
+        encoder_cw.pressed = false;
+        encoder_cw.time = (timer_read() | 1);
+        action_exec(encoder_cw);
+    }
+}
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise) {
+        encoder_cw.pressed = true;
+        encoder_cw.time = (timer_read() | 1);
+        action_exec(encoder_cw);
+    } else {
+        encoder_ccw.pressed = true;
+        encoder_ccw.time = (timer_read() | 1);
+        action_exec(encoder_ccw);
+    }
+}
```

Comment:
Oh my. That's cheating.
LOVE IT!

Reply:
Haha, can't take credit for this though. Not sure where this was first done, but I saw this here: https://github.com/NightlyBoards/qmk_firmware/commit/fe7949d6f5783c184f3e7d3ee4df7b4bb86f26ba